### PR TITLE
nfc: Update changelog for Developer 4 release.

### PR DESCRIPTION
**Pull request by grochu**
Updated changelog with version 0.3.0 of the NFC libraries.

@kapi-no  could you please review?

### CODE REVIEW

File: nfc/CHANGELOG.rst [hunk 1]
```diff
@@ -5,6 +5,24 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+NFC 0.3.0
+****************
+
+Fixed minor bugs in the NFC libraries.
+
+Modified
```
Comment (ru-fu) [inline nfc/CHANGELOG.rst]: Please change to:

```
Modified
========

* Modified the NFC platform module (``nfc_platform_zephyr.c``) to use the Zephyr Clock Controller driver.
* Fixed the behavior of the Type 2 Tag library (``libnfct2t_nrf52.a``) when receiving an unsupported WRITE command.

```

Comment (grochu) [reply nfc/CHANGELOG.rst]: Fixed. Thanks for reviewing.

**Comment by grochu** [comment]
Hi @lemrey could you review and merge?

**Comment by grochu** [comment]
Hi @pizi-nordic could you review/merge?

**Pull request merged**